src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -672,19 +673,29 @@ def get_active_daemon(self, daemon_descrs: List[DaemonDescription]) -> DaemonDes
         return DaemonDescription()
 
     def fail_over(self) -> None:
-        if not self.mgr_map_has_standby():
-            raise OrchestratorError('Need standby mgr daemon', event_kind_subject=(
-                'daemon', 'mgr' + self.mgr.get_mgr_id()))
-
-        self.mgr.events.for_daemon('mgr' + self.mgr.get_mgr_id(),
-                                   'INFO', 'Failing over to other MGR')
-        logger.info('Failing over to other MGR')
-
-        # fail over
-        ret, out, err = self.mgr.check_mon_command({
-            'prefix': 'mgr fail',
-            'who': self.mgr.get_mgr_id(),
-        })
+        # this has been seen to sometimes transiently fail even when there are multiple
+        # mgr daemons. As long as there are multiple known mgr daemons, we should retry.
+        class NoStandbyError(OrchestratorError):
+            pass
+        no_standby_exc = NoStandbyError('Need standby mgr daemon', event_kind_subject=(
+            'daemon', 'mgr' + self.mgr.get_mgr_id()))
+        for sleep_secs in [1, 3, 6]:
```

Comment:
(nit) wondering if we will need to wait for a longer duration than this?

Reply:
Upped it to 2, 8, 15. That's as long as I'd really like us to be waiting here as this will be holding up the serve loop when it's happening. I think 25 seconds is also long enough that we are justified marking the upgrade as failed.

Also added an info level log statement saying we're waiting to retry fail over so anyone looking at the logs will know why things aren't progressing for that 25 seconds.